ratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-73_R1_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-75_R1_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-76_R1_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-78_R1_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-81_R1_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-83_R1_001.fastq.gz.fastp-trim.20220810.fq.gz --right /gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-19_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-23_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-24_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-34_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-35_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-36_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-37_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-38_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-39_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-40_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-42_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-43_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-48_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-49_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-52_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-54_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-56_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-57_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-58_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-59_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-61_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-63_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-64_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-67_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-69_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-71_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-73_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-75_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-76_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-78_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-81_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,/gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-83_R2_001.fastq.gz.fastp-trim.20220810.fq.gz --pairs_together  --PARALLEL_STATS
-prepping seqs
Converting input files. (both directions in parallel)CMD: seqtk-trinity seq -A -R 1  <(gunzip -c /gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-19_R1_001.fastq.gz.fastp-trim.20220810.fq.gz) >> left.fa
CMD: seqtk-trinity seq -A -R 2  <(gunzip -c /gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-19_R2_001.fastq.gz.fastp-trim.20220810.fq.gz) >> right.fa
CMD finished (61 seconds)
CMD: seqtk-trinity seq -A -R 1  <(gunzip -c /gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-23_R1_001.fastq.gz.fastp-trim.20220810.fq.gz) >> left.fa
CMD finished (65 seconds)
CMD: seqtk-trinity seq -A -R 2  <(gunzip -c /gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-23_R2_001.fastq.gz.fastp-trim.20220810.fq.gz) >> right.fa
CMD finished (64 seconds)
CMD: seqtk-trinity seq -A -R 1  <(gunzip -c /gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-24_R1_001.fastq.gz.fastp-trim.20220810.fq.gz) >> left.fa
CMD finished (64 seconds)
CMD: seqtk-trinity seq -A -R 2  <(gunzip -c /gscratch/srlab/graceac9/analyses/pycno/20220810_PSC2021_trimming/PSC-24_R2_001.fastq.gz.fastp-trim.20220810.fq.gz) >> right.fa
q
^C
[graceac9@mox1 jobs]$ clear

[graceac9@mox1 jobs]$ nano 20220811_pycno_trinity_RNAseq_transcriptome.sh
















































  GNU nano 2.3.1                            File: 20220811_pycno_trinity_RNAseq_transcriptome.sh

${data_dir}/PSC-58_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,\
${data_dir}/PSC-59_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,\
${data_dir}/PSC-61_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,\
${data_dir}/PSC-63_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,\
${data_dir}/PSC-64_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,\
${data_dir}/PSC-67_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,\
${data_dir}/PSC-69_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,\
${data_dir}/PSC-71_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,\
${data_dir}/PSC-73_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,\
${data_dir}/PSC-75_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,\
${data_dir}/PSC-76_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,\
${data_dir}/PSC-78_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,\
${data_dir}/PSC-81_R2_001.fastq.gz.fastp-trim.20220810.fq.gz,\
${data_dir}/PSC-83_R2_001.fastq.gz.fastp-trim.20220810.fq.gz \
--CPU 28

# Assembly stats
${trinity_dir}/util/TrinityStats.pl \
trinity_out_dir/Trinity.fasta \
> ${assembly_stats}

# Create gene map files
${trinity_dir}/util/support_scripts/get_Trinity_gene_to_trans_map.pl \
trinity_out_dir/Trinity.fasta \
> trinity_out_dir/Trinity.fasta.gene_trans_map

# Create sequence lengths file (used for differential gene expression)
${trinity_dir}/util/misc/fasta_seq_length.pl \
trinity_out_dir/Trinity.fasta \
> trinity_out_dir/Trinity.fasta.seq_lens

# Create FastA index
${programs_array[samtools_faidx]} \
trinity_out_dir/Trinity.fasta
