---
title: "`DESeq2`with 2015Phel Transciptome and 2021PSC `kallisto` counts"
output: html_document
---

Rmd to get Differential gene expression information for 2021 _Pycnopodia helianthoides_ coelomocyte RNAseq library `kallisto` counts from comparing them to the 2015 _Pycnopodia helianthoides_ transcriptome.     [code/04-kallisto-2015Phel_transcriptome.Rmd](https://github.com/grace-ac/project_pycno/blob/main/code/04-kallisto-2015Phel_transcriptome.Rmd)

RNAseq library information for figuring out different libraries to compare: [PSC_sample_comparison_options google sheets](https://docs.google.com/spreadsheets/d/1LtIQ-VBCQNi1jJCGmf7DlA4f8Y410-CxwM1z6l-EjIY/edit#gid=0)

Load packages:
```{r}
library(DESeq2)
library(tidyverse)
library(pheatmap)
library(RColorBrewer)
library(data.table)
```


## Read in count matrix from comparing the 2021 _Pycnopodia helianthoides_ coelomocyte RNAseq libraries with the 2015 _Pycnopodia helianthoides_ transcriptome:  
```{r}
countmatrix <- read.delim("../analyses/Kallisto/2015Phel_transcriptome/kallisto-20220824_2015.isoform.counts.matrix", header = TRUE, sep = '\t')
rownames(countmatrix) <- countmatrix$X
countmatrix <- countmatrix[,-1]
head(countmatrix)
```
29476 obs, 33 variables (32 libraries)

Round integers up to whole numbers for analyses:    
```{r}
countmatrix <- round(countmatrix, 0)
head(countmatrix)
```

## Get DEGs based on healthy vs sick
Healthy Stars:   Library ID:
E06                 PSC.56
E07                 PSC.81
E09                 PSC.61
E10                 PSC.76
E11                 PSC.52
E14                 PSC.48
E16                 PSC.49
E18                 PSC.54

Sick Stars:       Libary ID:
H01                 PSC.59
H03                 PSC.63 (also PSC.73 a day after)
H04                 PSC.69
H05                 PSC.58 (also PSC.64 a day after)
H06                 PSC.57
H07                 PSC.83
H08                 PSC.75
H09                 PSC.67

### Need to subset the countmatrix for those 16 libraries:    
```{r}
counts_hvs <- countmatrix[,c("PSC.56","PSC.81","PSC.61","PSC.76","PSC.52","PSC.48","PSC.49","PSC.54","PSC.59","PSC.63","PSC.69","PSC.58","PSC.57","PSC.83","PSC.75","PSC.67")]
head(counts_hvs)
```
29476 rows, 16 columns (yay)


Make a data frame for the comparison: 
```{r}
colData <- data.frame(condition=factor(c("healthy","healthy","healthy","healthy","healthy","healthy","healthy","healthy","sick","sick","sick","sick","sick","sick","sick","sick")),
                      type=factor(rep("paired-end",16)))
rownames(colData) <- colnames(counts_hvs)
dds <- DESeqDataSetFromMatrix(countData = counts_hvs,
                              colData = colData,
                              design = ~ condition)
```

```{r}
dds <- DESeq(dds)
res <- results(dds)
res <- res[order(rownames(res)), ]
```

```{r}
head(res)
```

```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(res[!is.na(res$padj) & res$padj <= 0.05, ])
```
8981 rows 

```{r}
hvs_fig <- res
# The main plot
plot(hvs_fig$baseMean, hvs_fig$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     main="Healthy vs Sick  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
hvs_fig.sig <- res[!is.na(res$padj) & res$padj <= 0.05, ]
points(hvs_fig.sig$baseMean, hvs_fig.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```
1,861 <=0 omitted from logarithmic plot

Write out DEG file: 
```{r}
write.table(hvs_fig.sig, "../analyses/DESeq2/2015Phel/DEGlist_healthy-vs-sick.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
Wrote out table 20220907 @ 15:52. 









