---
title: "`DESeq2`with 2015Phel Transciptome and 2021PSC `kallisto` counts"
output: html_document
---

Rmd to get Differential gene expression information for 2021 _Pycnopodia helianthoides_ coelomocyte RNAseq library `kallisto` counts from comparing them to the 2015 _Pycnopodia helianthoides_ transcriptome.     [code/04-kallisto-2015Phel_transcriptome.Rmd](https://github.com/grace-ac/project_pycno/blob/main/code/04-kallisto-2015Phel_transcriptome.Rmd)

RNAseq library information for figuring out different libraries to compare: [RNAseq_library_metadata google sheet](https://docs.google.com/spreadsheets/d/1j1J63iS9lp_Ey5VqdNH_IzyM-uNS_5mvW71V4OxYtOI/edit#gid=0)

```{r}
sessionInfo()
```


```{r}
#if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

#BiocManager::install("DESeq2")
```

Load packages:
```{r}
library(DESeq2)
library(tidyverse)
library(pheatmap)
library(RColorBrewer)
library(data.table)
library(ggplot2)
```

```{r}
#packageVersion("rlang")
```

```{r}
#install.packages("rlang")
```

## Read in count matrix from comparing the 2021 _Pycnopodia helianthoides_ coelomocyte RNAseq libraries with the 2015 _Pycnopodia helianthoides_ transcriptome:  
```{r}
countmatrix <- read.delim("../analyses/Kallisto/2015Phel_transcriptome/kallisto-20220824_2015.isoform.counts.matrix", header = TRUE, sep = '\t')
rownames(countmatrix) <- countmatrix$X
countmatrix <- countmatrix[,-1]
head(countmatrix)
```
29476 obs, 32 variables (32 libraries)

Round integers up to whole numbers for analyses:    
```{r}
countmatrix <- round(countmatrix, 0)
head(countmatrix)
```

## Get DEGs based on healthy vs sick
NOTE: healthy samples are those that had no disease signs at or before sampling and no exposure to tank illness or a filtered live inoculate
Healthy : 

| Library ID | Star ID | Sample Date | Star Death Date | Disease Signs at Time of Sampling | Notes                                                  |
|------------|---------|-------------|-----------------|-----------------------------------|--------------------------------------------------------|
| PSC.58     | H05     | 2021-10-14  | 2021-10-18      | NA                                | star remained healthy; sacrificed at end of experiment |
| PSC.63     | H03     | 2021-10-15  | 2021-10-18      | NA                                | star remained healthy; sacrificed at end of experiment |
| PSC.64     | H05     | 2021-10-15  | 2021-10-18      | NA                                | star remained healthy; sacrificed at end of experiment |
| PSC.73     | H03     | 2021-10-16  | 2021-10-18      | NA                                | star remained healthy; sacrificed at end of experiment |


Note: Sick samples are those that were sampled at or during disease sign development and received heat-killed inoculate prior to experiment 2 and did not have exposure to the tank outbreak       
Sick :      
| Library ID | Star ID | Sample Date | Star Death Date | Disease Signs at Time of Sampling         |
|------------|---------|-------------|-----------------|-------------------------------------------|
| PSC.57     | H06     | 2021-10-14  | 2021-10-16      | 1 arm lost, twisting, lesion              |
| PSC.59     | H01     | 2021-10-14  | 2021-10-16      | 1 arm lost, twisting, lesions, stretching |
| PSC.67     | H09     | 2021-10-15  | 2021-10-16      | 2 arms lost, twisting                     |
| PSC.69     | H04     | 2021-10-15  | 2021-10-16      | 9 arms lost, twisting                     |

### 2022-12-02   
Need to subset the countmatrix for those 8 libraries:    
```{r}
counts_hvs <- countmatrix[,c("PSC.58", "PSC.63", "PSC.64", "PSC.73", "PSC.57", "PSC.59", "PSC.67", "PSC.69")]
head(counts_hvs)
```
29476 rows, 8 columns 


Make a data frame for the comparison: 
```{r}
colData <- data.frame(condition=factor(c("healthy","healthy","healthy","healthy","sick","sick","sick","sick")),
                      type=factor(rep("paired-end",8)))
rownames(colData) <- colnames(counts_hvs)
dds <- DESeqDataSetFromMatrix(countData = counts_hvs,
                              colData = colData,
                              design = ~ condition)
```


```{r}
dds <- DESeq(dds)
res <- results(dds)
res <- res[order(rownames(res)), ]
```

From [Bioconductor `DESeq2` Vignette](http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html):     
```{r}
vsd <- vst(dds, blind=FALSE)
rld <- rlog(dds, blind=FALSE)
head(assay(vsd), 3)
```
```{r}
plotPCA(vsd, intgroup=c("condition", "type"))
```


```{r}
head(res)
```

```{r}
# Count number of hits with adjusted p-value less than 0.05
dim(res[!is.na(res$padj) & res$padj <= 0.05, ])
```
7490 DEGs. 


```{r}
hvs_fig <- res
# The main plot
plot(hvs_fig$baseMean, hvs_fig$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     main="Healthy vs Sick  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
hvs_fig.sig <- res[!is.na(res$padj) & res$padj <= 0.05, ]
points(hvs_fig.sig$baseMean, hvs_fig.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```
Warning: 1980 x values <= 0 omitted from logarithmic plot

Write out DEG file: 
```{r}
#write.table(hvs_fig.sig, "../analyses/DESeq2/2015Phel/DEGlist_healthy-vs-sick.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
Wrote out table 20221204 @ 19:23. 

NOTE: this might not be super robust because the healthy libraries are from just 2 stars whereas sick are from 4 different stars. 

20221204: get table of DEGs from above comparison that are </= p-value 0.05 AND log2FC <-2.00 and >2.00. 

```{r}
# Try to get number of hits with adj-p-value less than 0.05; as well as those with log2FC >2.00 and <-2.00
# get those greater than 2.00 log2FC
hvs_fig.sig2 <- subset.data.frame(hvs_fig.sig, hvs_fig.sig$log2FoldChange > 2.00)
# get those less than -2.00 log2FC
hvs_fig.sig3 <- subset.data.frame(hvs_fig.sig, hvs_fig.sig$log2FoldChange < -2.00)
# combine the two into 1 list of DEGs with p-values </= 0.05 and log2FC >2.00 and <-2.00
hvs_fig.sig.final <- rbind(hvs_fig.sig2, hvs_fig.sig3)

#look at head of combined file:
head(hvs_fig.sig.final)

```
Write out this new file:
```{r}
#write.table(hvs_fig.sig.final, "../analyses/DESeq2/2015Phel/DEGlist_healthy-vs-sick_signif.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
wrote out 20221202. 3,138 DEGs

# Clear environment in R studio and re-import the count matrix above.

## Get DEGs based on healthy vs sick with more libraries per group (some of healthy had been previously inoculated with 0.45 filtered live, but NEVER showed any disease signs)

### Sick stars
| Library ID | Star ID | Sample Date | Star Death Date | Disease Signs at Time of Sampling         | Notes                                                                                                                                                                                             |
|------------|---------|-------------|-----------------|-------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| PSC.57     | H06     | 2021-10-14  | 2021-10-16      | 1 arm lost, twisting, lesion              | inoculated with heat-killed 20210923, inoculated with live unfiltered 20211005                                                                                                                    |
| PSC.59     | H01     | 2021-10-14  | 2021-10-16      | 1 arm lost, twisting, lesions, stretching | inoculated with heat-killed 20210923, inoculated with live unfiltered 20211005                                                                                                                    |
| PSC.67     | H09     | 2021-10-15  | 2021-10-16      | 2 arms lost, twisting                     | inoculated with heat-killed 20210923, inoculated with live unfiltered 20211005                                                                                                                    |
| PSC.69     | H04     | 2021-10-15  | 2021-10-16      | 9 arms lost, twisting                     | inoculated with heat-killed 20210923, inoculated with live unfiltered 20211005                                                                                                                    |
| PSC.71     | H18     | 2021-10-15  | 2021-10-15      | 14 arms lost, twisting                    | inoculated with heat-killed 20210923, inoculated with live unfiltered 20211005                                                                                                                    |
| PSC.75     | H08     | 2021-10-16  | 2021-10-18      | 1 arm lost, twisting                      | inoculated with heat-killed 20210923, inoculated with live unfiltered 20211005                                                                                                                    |
| PSC.78     | H10     | 2021-10-17  | 2021-10-18      | 5 arms lost, twisting                     | inoculated with heat-killed 20210923, inoculated with live unfiltered 20211005                                                                                                                    |
| PSC.83     | H07     | 2021-10-18  | 2021-10-18      | 6 arms twisted, stretching                | inoculated with heat-killed 20210923, inoculated with live unfiltered 20211005; we ended experiment 20211018, but star was showing definite signs of sickness despite not having dropped arms yet |

### healthy stars     
 Note: a couple had been inoculated with 0.45 filtered live inoc during experiment 1 (20210923 inoc. date), but never showed signs of disease. were later inoculated with heat-killed 20211005 for experiment 2. Also, used 2 libraries from star H05 and two time points in order to get 8 librariese total for even comparison. 

| Library ID | Star ID | Sample Date | Star Death Date | Disease Signs at Time of Sampling | Notes                                                                         |
|------------|---------|-------------|-----------------|-----------------------------------|-------------------------------------------------------------------------------|
| PSC.58     | H05     | 2021-10-14  | 2021-10-18      | NA                                | inoculated with heat-killed for exp 1 (20210923); never any disease signs     |
| PSC.63     | H03     | 2021-10-15  | 2021-10-18      | NA                                | inoculated with heat-killed 20210923; never any disease signs                 |
| PSC.52     | E11     | 2021-10-14  | 2021-10-18      | NA                                | inoculated with 0.45um filtered live 20210923, never showed any disease signs |
| PSC.54     | E18     | 2021-10-14  | 2021-10-18      | NA                                | inoculated with 0.45um filtered live 20210923, never showed any disease signs |
| PSC.61     | E09     | 2021-10-15  | 2021-10-18      | NA                                | inoculated with 0.45um filtered live 20210923, never showed any disease signs |
| PSC.76     | E10     | 2021-10-17  | 2021-10-18      | NA                                | inoculated with 0.45um filtered live 20210923, never showed any disease signs |
| PSC.81     | E07     | 2021-10-17  | 2021-10-18      | NA                                | inoculated with 0.45um filtered live 20210923, never showed any disease signs |
| PSC.64     | H05     | 2021-10-15  | 2021-10-18      | NA                                | inoculated with heat-killed 20210923 and never showed signs of disease        |

```{r}
counts_hvs2 <- countmatrix[,c("PSC.58", "PSC.63", "PSC.52", "PSC.54", "PSC.61", "PSC.76", "PSC.81", "PSC.64", "PSC.57", "PSC.59", "PSC.67", "PSC.69", "PSC.71", "PSC.75", "PSC.78", "PSC.83")]
head(counts_hvs2)
```
 
Make a data frame for the comparison: 
```{r}
colData <- data.frame(condition=factor(c("healthy","healthy","healthy","healthy","healthy", "healthy", "healthy" ,"healthy", "sick","sick","sick","sick", "sick", "sick", "sick", "sick")),
                      type=factor(rep("paired-end",16)))
rownames(colData) <- colnames(counts_hvs2)
dds <- DESeqDataSetFromMatrix(countData = counts_hvs2,
                              colData = colData,
                              design = ~ condition)
```
```{r}
dds <- DESeq(dds)
res <- results(dds)
res <- res[order(rownames(res)), ]
```
From [Bioconductor `DESeq2` Vignette](http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html):     
```{r}
vsd <- vst(dds, blind=FALSE)
rld <- rlog(dds, blind=FALSE)
head(assay(vsd), 3)
```
```{r}
plotPCA(vsd, intgroup=c("condition", "type"))
```


```{r}
plot <- plotPCA(vsd, intgroup=c("condition", "type"))
nudge <- position_nudge(y = 2)
plot + geom_text(aes(label = name), position = nudge)
```



```{r}
head(res)
```

```{r}
# Count number of hits with adjusted p-value less than 0.05
dim(res[!is.na(res$padj) & res$padj <= 0.05, ])
```
12424 degs

```{r}
hvs_fig <- res
# The main plot
plot(hvs_fig$baseMean, hvs_fig$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     main="Healthy vs Sick  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
hvs_fig.sig <- res[!is.na(res$padj) & res$padj <= 0.05, ]
points(hvs_fig.sig$baseMean, hvs_fig.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```

Warning: 1865 x values <= 0 omitted from logarithmic plot

Write out DEG file: 
```{r}
write.table(hvs_fig.sig, "../analyses/DESeq2/2015Phel/DEGlist_healthy-vs-sick_8x8.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
HUGE file
wrote out 20221204

```{r}
# Try to get number of hits with adj-p-value less than 0.05; as well as those with log2FC >2.00 and <-2.00
# get those greater than 2.00 log2FC
hvs_fig.sig2 <- subset.data.frame(hvs_fig.sig, hvs_fig.sig$log2FoldChange > 2.00)
# get those less than -2.00 log2FC
hvs_fig.sig3 <- subset.data.frame(hvs_fig.sig, hvs_fig.sig$log2FoldChange < -2.00)
# combine the two into 1 list of DEGs with p-values </= 0.05 and log2FC >2.00 and <-2.00
hvs_fig.sig.final <- rbind(hvs_fig.sig2, hvs_fig.sig3)

#look at head of combined file:
head(hvs_fig.sig.final)

```

Write out this new file:
```{r}
write.table(hvs_fig.sig.final, "../analyses/DESeq2/2015Phel/DEGlist_healthy-vs-sick_signif_8x8.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
3876 degs
wrote out 20221204

# Clear Rstudio environment

## Get DEGs when comparing counts between 4 libraries of stars that died after tank exposure before experiment 2 start to 4 libraries of stars that  died in experiment 2 after inoculation with live unfiltered inoculum (were previously injected with heat-killed in experiment 1)

### libraries of the 4 stars with tank exposure that died early in experiment 1 (likely died from tank exposure). 

| Library ID | Star ID | Sample Date | Star Death Date | Disease Signs at Time of Sampling | Notes                                                                                                          |
|------------|---------|-------------|-----------------|-----------------------------------|----------------------------------------------------------------------------------------------------------------|
| PSC.19     | H16     | 2021-09-23  | 2021-09-24      | 2 arms lost, lesions, stretching  | had tank exposure (date of tank outbreak: ~20210908); inoculated with heat-killed for experiment 1 20210923    |
| PSC.23     | H15     | 2021-09-26  | 2021-09-27      | 2 arms lost, lesions              | had tank exposure (date of tank outbreak: ~20210908); inoculated with heat-killed for experiment 1 20210923    |
| PSC.24     | E13     | 2021-09-27  | 2021-09-28      | 6 arms lost                       | had tank exposure (date of tank outbreak: ~20210908); inoculated with 0.45um filtered live for exp. 1 20210923 |
| PSC.34     | H12     | 2021-10-02  | 2021-10-02      | 11 arms lost                      | had tank exposure (date of tank outbreak: ~20210908); inoculated with heat-killed for experiment 1 20210923    |

### libraries of 4 sick stars from experiment 2 that were injected with heat-killed for experiemnt 1, then live unfiltered for experiment 2]]

| Library ID | Star ID | Sample Date | Star Death Date | Disease Signs at Time of Sampling         |
|------------|---------|-------------|-----------------|-------------------------------------------|
| PSC.57     | H06     | 2021-10-14  | 2021-10-16      | 1 arm lost, twisting, lesion              |
| PSC.59     | H01     | 2021-10-14  | 2021-10-16      | 1 arm lost, twisting, lesions, stretching |
| PSC.67     | H09     | 2021-10-15  | 2021-10-16      | 2 arms lost, twisting                     |
| PSC.69     | H04     | 2021-10-15  | 2021-10-16      | 9 arms lost, twisting   

Get counts for libraries of interest:
```{r}
counts_tvs <- countmatrix[,c("PSC.19", "PSC.23", "PSC.24", "PSC.34", "PSC.57", "PSC.59", "PSC.67", "PSC.69")]
head(counts_tvs)
```

Make a data frame for the comparison: 
```{r}
colData <- data.frame(condition=factor(c("tankexposed","tankexposed","tankexposed","tankexposed", "sick","sick","sick","sick")),
                      type=factor(rep("paired-end",8)))
rownames(colData) <- colnames(counts_tvs)
dds <- DESeqDataSetFromMatrix(countData = counts_tvs,
                              colData = colData,
                              design = ~ condition)
```

```{r}
dds <- DESeq(dds)
res <- results(dds)
res <- res[order(rownames(res)), ]
```

From [Bioconductor `DESeq2` Vignette](http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html):     
```{r}
vsd <- vst(dds, blind=FALSE)
rld <- rlog(dds, blind=FALSE)
head(assay(vsd), 3)
```

```{r}
plotPCA(vsd, intgroup=c("condition", "type"))
```

```{r}
head(res)
```

```{r}
# Count number of hits with adjusted p-value less than 0.05
dim(res[!is.na(res$padj) & res$padj <= 0.05, ])
```
120 degs 

```{r}
tvs_fig <- res
# The main plot
plot(tvs_fig$baseMean, tvs_fig$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     main="Tank exposed vs Sick  (pval </= 0.05) ",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
tvs_fig.sig <- res[!is.na(res$padj) & res$padj <= 0.05, ]
points(tvs_fig.sig$baseMean, tvs_fig.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```
Warning: 1977 x values <= 0 omitted from logarithmic plot

Write out DEG file: 
```{r}
#write.table(tvs_fig.sig, "../analyses/DESeq2/2015Phel/DEGlist_tankexposed_v_sick_4x4_pval.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
wrote out 20221204

```{r}
# Try to get number of hits with adj-p-value less than 0.05; as well as those with log2FC >2.00 and <-2.00
# get those greater than 2.00 log2FC
tvs_fig.sig <- subset.data.frame(tvs_fig.sig, tvs_fig.sig$log2FoldChange > 2.00)
# get those less than -2.00 log2FC
tvs_fig.sig2 <- subset.data.frame(tvs_fig.sig, tvs_fig.sig$log2FoldChange < -2.00)
# combine the two into 1 list of DEGs with p-values </= 0.05 and log2FC >2.00 and <-2.00
tvs_fig.sig.final <- rbind(tvs_fig.sig, tvs_fig.sig2)

#look at head of combined file:
head(tvs_fig.sig.final)

```
Write out this new file:
```{r}
write.table(tvs_fig.sig.final, "../analyses/DESeq2/2015Phel/DEGlist_tankexposed_v_sick_4x4.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
15 degs
wrote out 20221204


