---
title: "11.0.0-trinity-denovo-unmapped-reads"
author: "Sam White"
date: "2024-05-30"
output: 
  bookdown::html_document2:
    theme: cosmo
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
    code_download: true
  github_document:
    toc: true
    number_sections: true
  html_document:
    theme: cosmo
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
    code_download: true
bibliography: references.bib
link-citations: true
---

Trinity _de novo_ assembly of unmapped, HiSat2 reads, per [this GitHub Issue](https://github.com/RobertsLab/resources/issues/1915).



```{r setup, include=FALSE}
library(reticulate)
library(knitr)
library(dplyr)
knitr::opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = FALSE,        # Evaluate code chunks
  warning = FALSE,     # Hide warnings
  message = FALSE,     # Hide messages
  comment = ""         # Prevents appending '##' to beginning of lines in code output
)
```


# Set R variables

Needed for Python usage by Trinity modules
```{r R-variables, eval=TRUE}
conda_env_name <- c("base")
conda_path <- c("/home/sam/programs/mambaforge/bin/conda")
```

# Create a Bash variables file

This allows usage of Bash variables across R Markdown chunks.

```{r save-bash-variables-to-rvars-file, engine='bash', eval=TRUE}
{
echo "#### Assign Variables ####"
echo ""

echo "# Data directories"
echo ""
echo "# RNAseq FastQs directory"
echo 'export reads_dir="/home/shared/8TB_HDD_01/sam/gitrepos/project-pycno-sswd-2021/data"'
echo ""

echo "# FastQ URL"
echo 'export url="https://gannet.fish.washington.edu/seashell/bu-github/paper-pycno-sswd-2021-2022/analyses/10-hisat-deseq2/"'

echo "# Output directories"
echo 'export output_dir="/home/shared/8TB_HDD_01/sam/gitrepos/project-pycno-sswd-2021/analyses/11.0.0-trinity-denovo-unmapped-reads"'
echo 'export trinity_out_dir="${output_dir}/trinity_out_dir"'
echo ""

echo "# Set FastQ filename patterns"
echo "export fastq_pattern='*.fastq'"
echo "export R1_fastq_pattern='*1.fastq'"
echo "export R2_fastq_pattern='*2.fastq'"

echo "# Inititalize arrays"
echo "# Leave empty!!"
echo 'export R1_array=()'
echo 'export R2_array=()'
echo ""

echo "# Variables for R1/R2 lists"
echo "# Leave empty!!!"
echo 'export R1_list=""'
echo 'export R2_list=""'
echo ""

echo "# CPU threads"
echo 'export threads=40'
echo ""

echo "# Recommended maximum memory is 100GB, per Trinity developer"
echo 'export max_mem=100G'
echo ""

echo "# Name output files"
echo 'export fasta_name="pcoe-unmapped-de_novo-transcriptome_v1.0.fasta"'
echo 'export assembly_stats="${fasta_name}_assembly_stats.txt"'
echo ""

echo "# Paths to programs"
echo 'export samtools="/home/shared/samtools-1.12samtools"'
echo 'export trinity_dir="/home/shared/trinityrnaseq-v2.12.0"'
echo 'export trinity=${trinity_dir}/Trinity'
echo ""

echo "# Trinity needs following programs in system $PATH"
echo 'export PATH=${PATH}:/home/shared/bowtie2-2.4.4-linux-x86_64/'
echo 'export PATH=${PATH}:/home/shared/jellyfish-2.3.0/bin/'
echo 'export PATH=${PATH}:/home/shared/salmon-1.4.0_linux_x86_64/bin/'
echo ""

} > .bashvars

cat .bashvars
```

# Load conda environment

If this is successful, the first line of output should show that the Python being used is the one in your conda environment path.

E.g.

`python:         /home/sam/programs/mambaforge/envs/mirmachine_env/bin/python`

```{r load-conda-env, eval=TRUE}
use_condaenv(condaenv = conda_env_name, conda = conda_path)

# Check successful env loading
py_config()
```

# Download PSC-34_unmapped_reads*

```{r download-Pmea-genome-fasta, engine='bash', eval=TRUE}
# Load bash variables into memory
source .bashvars

wget \
--directory-prefix ${reads_dir} \
--recursive \
--no-check-certificate \
--continue \
--no-host-directories \
--no-directories \
--no-parent \
--quiet \
--execute robots=off \
--accept-regex "PSC-34_unmapped_reads\.[12]\.fastq" ${url}

ls -lh "${reads_dir}"
```

# ASSEMBLY

```{r assembly, engine='bash', eval=FALSE}
# Load bash variables into memory
source .bashvars

# Make output directory, if it doesn't exist
mkdir --parents "${output_dir}"

# Change to output directory
cd "${output_dir}"


# Programs associative array
declare -A programs_array
programs_array=(
[samtools_faidx]="${samtools} faidx" \
[trinity]="${trinity}"
)

# Create array of fastq R1 files
# Set filename pattern
R1_array=("${reads_dir}"/${R1_fastq_pattern})

# Create array of fastq R2 files
# Set filename pattern
R2_array=("${reads_dir}"/${R2_fastq_pattern})

# Create list of fastq files used in analysis
## Uses parameter substitution to strip leading path from filename
for fastq in "${!R1_array[@]}"
do
  {
    md5sum "${R1_array[${fastq}]}"
    md5sum "${R2_array[${fastq}]}"
  } >> input_fastqs.md5
done

# Create comma-separated lists of FastQ reads
R1_list=$(echo "${R1_array[@]}" | tr " " ",")
R2_list=$(echo "${R2_array[@]}" | tr " " ",")


# Run Trinity
## Running as "stranded" (--SS_lib_type)
${programs_array[trinity]} \
--seqType fq \
--SS_lib_type RF \
--max_memory ${max_mem} \
--CPU ${threads} \
--left "${R1_list}" \
--right "${R2_list}" \
&> trinity.log

# Rename generic assembly FastA
find . -name "Trinity*.fasta" -exec mv {} "${trinity_out_dir}"/"${fasta_name}" \;

# Assembly stats
${trinity_dir}/util/TrinityStats.pl "${trinity_out_dir}"/"${fasta_name}" \
> ${assembly_stats}

# Create gene map files
${trinity_dir}/util/support_scripts/get_Trinity_gene_to_trans_map.pl \
"${trinity_out_dir}"/"${fasta_name}" \
> "${fasta_name}".gene_trans_map

# Create sequence lengths file (used for differential gene expression)
${trinity_dir}/util/misc/fasta_seq_length.pl \
"${trinity_out_dir}"/"${fasta_name}" \
> "${fasta_name}".seq_lens

# Move FastA to working directory
mv "${trinity_out_dir}"/"${fasta_name}" .

# Create FastA index
${programs_array[samtools_faidx]} \
"${fasta_name}"

# Copy files to transcriptome directory
rsync -av \
"${fasta_name}"* \
"${output_dir}"

# Cleanup
rm -rf trinity_out_dir/

# Generate MD5 checksums
for file in *
do
  echo ""
  echo "Generating MD5 checksums for ${file}:"
  md5sum "${file}" | tee --append checksums.md5
  echo ""
done

```

# CHECK OUTPUTS

## Assembly Stats
```{r transcriptome, engine='bash', eval=TRUE}
# Load bash variables into memory
source .bashvars

cat "${output_dir}"/${assembly_stats}
```