12.0.0-blastn-unmapped-assembly
================
Sam White
2024-06-03

- <a href="#1-create-a-bash-variables-file"
  id="toc-1-create-a-bash-variables-file">1 Create a Bash variables
  file</a>
- <a href="#2-run-blastn" id="toc-2-run-blastn">2 Run BLASTn</a>
- <a href="#3-inspect-output" id="toc-3-inspect-output">3 Inspect
  output</a>
  - <a href="#31-preview-blast-output" id="toc-31-preview-blast-output">3.1
    Preview BLAST output</a>
  - <a href="#32-count-species" id="toc-32-count-species">3.2 Count
    species</a>

Run NCBI BLASTn ([Camacho et al. 2009](#ref-camacho2009)) on [Trinity
*de novo* assembly of unmapped, HiSat2
reads](./11.0.0-trinity-denovo-unmapped-reads.Rmd) using NCBI `nr`
database.

NCBI `nt` BLAST database was downloaded 20240603 by Sam White, using the
following command:

``` bash
update_blastdb.pl --num_threads 0 --decompress nt
```

# 1 Create a Bash variables file

This allows usage of Bash variables across R Markdown chunks.

``` bash
{
echo "#### Assign Variables ####"
echo ""

echo "# BLAST Database Directory"
echo 'export blast_dbs_dir="/home/shared/8TB_HDD_01/sam/data/blastdbs/20240603-nt"'
echo 'export BLASTDB="$BLASTDB:${blast_dbs_dir}"'
echo ""

echo "# Input files"
echo 'export blast_db="nt"'
echo 'export query_fasta="../analyses/11.0.0-trinity-denovo-unmapped-reads/pcoe-unmapped-de_novo-transcriptome_v1.0.fasta"'
echo ""

echo "# Output files"
echo 'export output_dir="../analyses/12.0.0-blastn-unmapped-assembly"'
echo 'export blast_output="pcoe-unmapped-de_novo-blastn.outfmt6"'
echo ""


echo "# Paths to programs"
echo 'export ncbi_blast_dir="/home/shared/ncbi-blast-2.15.0+/bin/"'
echo 'export ncbi_blastn="${ncbi_blast_dir}/blastn"'
echo ""

echo "# Set number of CPUs to use"
echo 'export threads=35'
echo ""


} > .bashvars

cat .bashvars
```

    #### Assign Variables ####

    # BLAST Database Directory
    export blast_dbs_dir="/home/shared/8TB_HDD_01/sam/data/blastdbs/20240603-nt"
    export BLASTDB="$BLASTDB:${blast_dbs_dir}"

    # Input files
    export blast_db="nt"
    export query_fasta="../analyses/11.0.0-trinity-denovo-unmapped-reads/pcoe-unmapped-de_novo-transcriptome_v1.0.fasta"

    # Output files
    export output_dir="../analyses/12.0.0-blastn-unmapped-assembly"
    export blast_output="pcoe-unmapped-de_novo-blastn.outfmt6"

    # Paths to programs
    export ncbi_blast_dir="/home/shared/ncbi-blast-2.15.0+/bin/"
    export ncbi_blastn="${ncbi_blast_dir}/blastn"

    # Set number of CPUs to use
    export threads=35

# 2 Run BLASTn

Look for top match (`-max_hsps 1` & `-max_target_seqs 1`) for each
query.

- Suppress subsequent warning
  `Examining 5 or more matches is recommended` by redirecting stdout:
  `2>blast.log`

- Uses customize output format 6 to include taxonomic IDs and scientific
  names.

``` bash
# Load bash variables into memory
source .bashvars

# Make output directory, if it doesn't exit
mkdir --parents "${blast_output}"

# Run BLASTn
${ncbi_blastn} \
-db ${blast_db} \
-query ${query_fasta} \
-out ${output_dir}/${blast_output} \
-max_hsps 1 \
-max_target_seqs 1 \
-outfmt "6 qaccver saccver ssciname staxid pident length mismatch gapopen qstart qend sstart send
   evalue bitscore" \
-num_threads ${threads} \
2> ${output_dir}/blast.log
```

# 3 Inspect output

## 3.1 Preview BLAST output

``` bash
# Load bash variables into memory
source .bashvars

head ${output_dir}/${blast_output}
```

    TRINITY_DN45185_c0_g1_i1    AH012920.2  uncultured Acidobacteriota bacterium    171953  95.652  276 11  1   1   275 29001   2.59e-119   442
    TRINITY_DN45142_c0_g1_i1    OU452219.1  Marthasterias glacialis 7609    98.148  54  1   0   2   55  14239465    5.76e-15    95.3
    TRINITY_DN45153_c0_g1_i1    NM_024773.3 Homo sapiens    9606    100.000 273 0   0   1   273 1160    3.23e-138   505
    TRINITY_DN45115_c0_g1_i1    CP134957.1  Asterias amurensis  7602    78.261  253 33  16  1   250 1995219 3.68e-29    143
    TRINITY_DN45136_c0_g1_i1    XM_006511516.4  Mus musculus    10090   100.000 296 0   0   1   296 3155    5.80e-151   547
    TRINITY_DN45133_c0_g1_i1    AF190464.1  Homo sapiens    9606    100.000 213 0   0   1   213 149404  5.44e-105   394
    TRINITY_DN45143_c0_g1_i1    LR699092.1  Asterias rubens 7604    77.919  394 50  23  10  383 10670971    1.21e-49    211
    TRINITY_DN45190_c0_g1_i1    NM_001009566.3  Homo sapiens    9606    100.000 364 0   0   1   364 4086    0.0 673
    TRINITY_DN45147_c0_g1_i1    NM_001311074.2  Mus musculus    10090   100.000 246 0   0   2   247 2976    2.93e-123   455
    TRINITY_DN45189_c0_g1_i1    NM_014188.3 Homo sapiens    9606    100.000 247 0   0   1   247 194 8.16e-124   457

## 3.2 Count species

``` bash
# Load bash variables into memory
source .bashvars

awk -F"\t" '{print $3}' ${output_dir}/${blast_output} \
| sort \
| uniq --count \
| sort --numeric-sort --reverse |
head
```

      17617 Mus musculus
      12185 Homo sapiens
       7251 Marthasterias glacialis
       6956 Asterias rubens
       2496 Asterias amurensis
        853 Vibrio pectenicida
        479 eukaryotic synthetic construct
        396 Mus caroli
        369 Pan troglodytes
        330 synthetic construct

<div id="refs" class="references csl-bib-body hanging-indent">

<div id="ref-camacho2009" class="csl-entry">

Camacho, Christiam, George Coulouris, Vahram Avagyan, Ning Ma, Jason
Papadopoulos, Kevin Bealer, and Thomas L Madden. 2009. “BLAST+:
Architecture and Applications.” *BMC Bioinformatics* 10 (1).
<https://doi.org/10.1186/1471-2105-10-421>.

</div>

</div>
