---
title: "20230522-for-prwg-mtg"
output: html_document
date: "2023-05-23"
---
Rmd for compiling just the results I'll share on Thursday May 25, 2023 at PRWG meeting. 

Goal of what to share:
1. DEGs of healthy vs sick (all samples are ~1.5weeks after exp. start date)
Healthy PSC: 58, 63, 61, 52, 54
Sick PSC: 59, 69, 67, 78, 71
1a. heatmap showing differences in expression between these groups
1b. PCA of these libraries showing how similar they are to each other
1c. DEG figure... but maybe it's not super interesting compared to heatmap
2. Specific cool genes: any anti-viral? anti-bacterial? immune-related?
2a. Include interpretation of what that might mean
3. Pull out those genes and follow the pattern within individual star over time, if found. 
3a. can follow 2 time points of sick star (H04), or healthy stars H03 or H05. 

Load packages:
```{r}
library(DESeq2)
library(tidyverse)
library(pheatmap)
library(RColorBrewer)
library(data.table)
library(ggplot2)
```


## Read in count matrix from comparing the 2021 _Pycnopodia helianthoides_ coelomocyte RNAseq libraries with the 2015 _Pycnopodia helianthoides_ transcriptome:  
```{r}
countmatrix <- read.delim("../analyses/Kallisto/2015Phel_transcriptome/kallisto-20220824_2015.isoform.counts.matrix", header = TRUE, sep = '\t')
rownames(countmatrix) <- countmatrix$X
countmatrix <- countmatrix[,-1]
head(countmatrix)
```

Round integers up to whole numbers for analyses:    
```{r}
countmatrix <- round(countmatrix, 0)
head(countmatrix)
```

## Get DEGs based on healthy vs sick
###   
Need to subset the countmatrix for those 12 libraries:    
```{r}
counts_hvs <- countmatrix[,c("PSC.58", "PSC.63", "PSC.61", "PSC.52", "PSC.54", "PSC.59", "PSC.69", "PSC.67", "PSC.78", "PSC.71")]
head(counts_hvs)
```

Make a data frame for the comparison: 
```{r}
colData <- data.frame(condition=factor(c("control","control","control", "control", "control", "exposed","exposed","exposed","exposed", "exposed")),
                      type=factor(rep("paired-end",10)))
rownames(colData) <- colnames(counts_hvs)
dds <- DESeqDataSetFromMatrix(countData = counts_hvs,
                              colData = colData,
                              design = ~ condition)
```
```{r}
dds <- DESeq(dds)
res <- results(dds)
res <- res[order(rownames(res)), ]
```

From [Bioconductor `DESeq2` Vignette](http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html):     
```{r}
vsd <- vst(dds, blind=FALSE)
rld <- rlog(dds, blind=FALSE)
head(assay(vsd), 3)
```

```{r}
plot <- plotPCA(vsd, intgroup=c("condition", "type"))
nudge <- position_nudge(y = 4)
plot + geom_text(aes(label = name), position = nudge)
```

```{r}
head(res)
```

```{r}
# Count number of hits with adjusted p-value less than 0.05
dim(res[!is.na(res$padj) & res$padj <= 0.05, ])
```
9936

```{r}
hvs_fig <- res
# The main plot
plot(hvs_fig$baseMean, hvs_fig$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     main="Control vs Exposed  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
hvs_fig.sig <- res[!is.na(res$padj) & res$padj <= 0.05, ]
points(hvs_fig.sig$baseMean, hvs_fig.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```

Warning: 1958 x values <= 0 omitted from logarithmic plot

Write out DEG file: 
```{r}
write.table(hvs_fig.sig, "../analyses/DESeq2/2015Phel/20230523_DEGlist_healthy-vs-sick_5x5.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
wrote out 20230523

```{r}
# Try to get number of hits with adj-p-value less than 0.05; as well as those with log2FC >2.00 and <-2.00
# get those greater than 2.00 log2FC
hvs_fig.sig2 <- subset.data.frame(hvs_fig.sig, hvs_fig.sig$log2FoldChange > 2.00)
# get those less than -2.00 log2FC
hvs_fig.sig3 <- subset.data.frame(hvs_fig.sig, hvs_fig.sig$log2FoldChange < -2.00)
# combine the two into 1 list of DEGs with p-values </= 0.05 and log2FC >2.00 and <-2.00
hvs_fig.sig.final <- rbind(hvs_fig.sig2, hvs_fig.sig3)

#look at head of combined file:
head(hvs_fig.sig.final)

```
Write out this new file:
```{r}
write.table(hvs_fig.sig.final, "../analyses/DESeq2/2015Phel/20230523-DEGlist_healthy-vs-sick_signif_5x5.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
wrote out 20230523

# `join` deg lists with library counts
load the 20230523-DEGlist_healthy-vs-sick_signif_5x5.tab
```{r}
deglist <- read.delim("../analyses/DESeq2/2015Phel/20230523-DEGlist_healthy-vs-sick_signif_5x5.tab")
head(deglist)
```



`join` with the count data from the libraries i compared:
make row names for both files into a column called "contig"
```{r}
counts_hvs <- rownames_to_column(counts_hvs, var = "contig")
head(counts_hvs)
```
```{r}
deglist <- rownames_to_column(deglist, var = "contig")
head(deglist)
```

```{r}
deglistcount <- left_join(deglist, counts_hvs, by = "contig")
head(deglistcount)
```
write out file:

```{r}
write.table(deglistcount, "../analyses/DESeq2/2015Phel/20230523-DEGlist_healthy-vs-sick_signif_5x5_counts.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
wrote out 20230523


# MAKE HEATMAP
Load in packages:
```{r}
library(tidyverse)
library(pheatmap)
library(RColorBrewer)
library(data.table)
```

Set heatmap colors:
```{r}
heatmapPurpleOrange <- RColorBrewer::brewer.pal(11, "PuOr")
```

Set Trinity_ID column as rownames:
```{r}
rownames(deglistcount) <- deglistcount$contig #set Trinity IDs as rownames
hvsheatmap <- deglistcount[,-1] #remove redundant column
head(hvsheatmap)
```

Pull out just the sample count data:
```{r}
hvsheatmap <- select(hvsheatmap, "PSC.58", "PSC.63", "PSC.61", "PSC.52", "PSC.54", "PSC.59", "PSC.69", "PSC.67", "PSC.78", "PSC.71" )
head(hvsheatmap)
```


Going to remove all rows that have "0" for all samples: 
```{r}
noz_hvsheatmap <- hvsheatmap[rowSums(hvsheatmap[, -1] >0) !=0, ]
head(noz_hvsheatmap)
```

```{r}
pheatmap(noz_hvsheatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 15, fontsize_row = 12)
```

Annotate heatmap with healthy or sick:
Add column annotations for the samples denoting temperature treatment and infection status:
```{r}
sample_col <- data.frame(treatment = c("control", "control", "control", "control", "control", "exposed", "exposed", "exposed", "exposed", "exposed"))
row.names(sample_col) <-colnames(noz_hvsheatmap)
sample_col
```


specify colors for annotation columns:
```{r}
my_colors <- list(
  treatment = c(control = "cyan3", exposed = "coral3")
)
```

```{r}
pheatmap(noz_hvsheatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", annotation_colors = my_colors, annotation_col = sample_col, show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 12, fontsize_row = 12)
```

cluster the heatmap
Set heatmap for temperature contrast as an object:
```{r}
results <- pheatmap(noz_hvsheatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 12, fontsize_row = 12)
```

Extract 4 clusters:
```{r}
contrast.clust <- cbind(noz_hvsheatmap,
                   cluster = cutree(results$tree_row,
                                     k = 4))
head(contrast.clust)
```

Pull out just the column of cluster numbers:
```{r}
clusters <- select(contrast.clust, "cluster")
head(clusters)
```

Rename the clusters in the rows to "cluster #" instead of just the number. 
```{r}
clusters <- data.frame(cluster = ifelse(clusters == 1, "cluster1", ifelse(clusters == 2, "cluster2", ifelse(clusters == 3, "cluster3", "cluster4"))))
head(clusters)
```

Add column annotations for the samples denoting health status:
```{r}
sample_col <- data.frame(health_status = c("healthy", "healthy", "healthy", "healthy", "healthy", "sick", "sick", "sick", "sick", "sick"))
row.names(sample_col) <-colnames(noz_hvsheatmap)
sample_col
```

my_colors <- list(
  health_status = c(healthy = "chartreuse4", sick = "coral3")
)
specify colors for annotation columns:
```{r}
my_colors <- list(
  health_status = c(healthy = "cyan3", sick = "coral3"),
  cluster = c(cluster1 = "#1B9E77", cluster2 = "#D95F02", cluster3 = "#7570B3", cluster4 = "#E7298A")
)
```

```{r}
pheatmap(noz_hvsheatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", annotation_colors = my_colors, annotation_row = clusters, annotation_col = sample_col, show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 12, fontsize_row = 12)
```


Set rownames in heatmap matrix as column called "Phel_contig":
```{r}
head(contrast.clust)
```


```{r}
contrast.clust <- tibble::rownames_to_column(contrast.clust, "Phel_contig")
head(contrast.clust)
```


`join` the deglist cluster list with annotation:

```{r}
phelannot <- read.delim("../analyses/DESeq2/2015Phel/Blastquery-GOslim.tab", sep = '\t', header = FALSE)
head(phelannot)
```

```{r}
cols <- c("Phel_contig", "GO_ID", "GO_slim", "CC_MF_BP")
colnames(phelannot) <- cols
head(phelannot)
```

```{r}
deglist_heatmap.go <- left_join(contrast.clust, phelannot, by = "Phel_contig")
head(deglist_heatmap.go)
```

write out table:
```{r}
write.table(deglist_heatmap.go, "../analyses/DESeq2/2015Phel/20230523_DEGs-clusters-annot.tab", sep = '\t', quote = FALSE, row.names = FALSE)
```
Wrote out 20230523. Commented out code. 


`join` _blast_annot.tab with the 20230523_DEGs-clusters-annot.tab 
```{r}
blastannot <- read.delim("../analyses/DESeq2/2015Phel/_blast-annot.tab", sep = '\t', header = FALSE)
head(blastannot)
```
Name column V2 Phel_contig:
```{r}
colsblast <- c("V1", "Phel_contig", "V3", "V4", "V5", "V6", "V7", "V8", "V9", "V10", "V11", "V12", "V13", "V14")
colnames(blastannot) <- colsblast
head(blastannot)
```

```{r}
degs.count.annot <- read.delim("../analyses/DESeq2/2015Phel/20230523_DEGs-clusters-annot.tab", sep = '\t', header = TRUE)
head(degs.count.annot)
```

```{r}
fullannot.degs <- left_join(degs.count.annot, blastannot, by = "Phel_contig")
head(fullannot.degs)
```
3651 rows, 30 columns

write out
```{r}
write.table(fullannot.degs, "../analyses/DESeq2/2015Phel/20230523-annotated_deglist-healthyVsick.tab", sep = "\t", row.names = F, quote = FALSE, col.names = TRUE)
```
wrote out 20230523

```{r}
highsick <- read.delim("../analyses/DESeq2/2015Phel/sick_20230524_phel_DAVIDout.rtf")
head(highsick)
```

# get unique contigs of degs to get number within sick and low sick groups:
```{r}
degshvs.filt <- read.delim("../analyses/DESeq2/2015Phel/20230523-annotated_deglist-healthyVsick.tab")
head(degshvs.filt)
```
```{r}
vignette("DESeq2")
```

